sdk/src/androidTest/java/com/mongodb/stitch/StitchClientTest.kt

```diff
@@ -0,0 +1,218 @@
+import com.mongodb.stitch.android.StitchClient
+import com.mongodb.stitch.android.auth.anonymous.AnonymousAuthProvider
+import com.mongodb.stitch.android.auth.oauth2.facebook.FacebookAuthProviderInfo
+import com.mongodb.stitch.android.auth.oauth2.google.GoogleAuthProviderInfo
+import io.appflate.restmock.RESTMockServer
+import org.junit.Before
+import org.junit.Test
+import org.junit.runner.RunWith
+import kotlin.reflect.full.declaredMemberProperties
+import kotlin.reflect.jvm.isAccessible
+import io.appflate.restmock.utils.RequestMatchers.pathContains
+import org.json.JSONObject
+import io.appflate.restmock.android.AndroidLogger
+import io.appflate.restmock.android.AndroidAssetsFileParser
+import io.appflate.restmock.RESTMockServerStarter
+import io.appflate.restmock.utils.RequestMatchers.pathEndsWith
+
+/**
+ * Test public methods of StitchClient, running through authentication
+ * and pipeline flows.
+ */
+@RunWith(AndroidJUnit4::class)
+class StitchClientTest {
+    companion object {
+        const val FakeAccessToken = "fake-access-token"
+        const val FakeRefreshToken = "fake-refresh-token"
+        const val FakeUserId = "fake-user-id"
+        const val FakeDeviceId = "fake-device-id"
+
+        const val FakeDeferUrl = "https://mock.com"
+        const val FakeGoogleClientId = "fake-google-client-id"
+        const val FakeFacebookClientId = "fake-facebook-client-id"
+        const val FakeMetadataFieldEmail = "email"
+        const val FakeMetadataFieldBirthday = "birthday"
+
+        const val FakePipelineLiteralFoo = "foo"
+        const val FakePipelineLiteralBar = "bar"
+
+        const val FakeDomainId = "fake-domain-id"
+        const val FakeAnonIdentity = "anon/user"
+        const val FakeUserType = "normal"
+
+        /** Mock data for available providers [StitchClient.getAuthProviders] */
+        val mockProviderData = JSONObject(mapOf(
+                "anon/user" to emptyMap<String, String>(),
+                "local/userpass" to mapOf(
+                        "emailConfirmationUrl" to FakeDeferUrl,
+                        "resetPasswordUrl" to FakeDeferUrl
+                ),
+                "oauth2/google" to mapOf(
+                        "clientId" to FakeGoogleClientId,
+                        "metadataFields" to arrayOf(
+                                FakeMetadataFieldEmail,
+                                FakeMetadataFieldBirthday
+                        )
+                ),
+                "oauth2/facebook" to mapOf(
+                        "clientId" to FakeFacebookClientId,
+                        "metadataFields" to arrayOf(
+                                FakeMetadataFieldEmail,
+                                FakeMetadataFieldBirthday
+                        )
+                )
+        )).toString()
+
+        /** Mock data for auth object from [StitchClient.logInWithProvider] */
+        val mockAuthData = JSONObject(mapOf(
+                "accessToken" to FakeAccessToken,
```

Comment:
We could, but they would be unused at the moment, as we aren't going to be accessing the mock data by keys, but in their mapped object form.